### PR TITLE
Use the associated program to open files by default.

### LINKED ISSUE #6: Using the program associated with the selected file(s) should be the default behaviour
Currently, the user has to manually pick an executable to launch the selected file(s).
This can be improved upon by applying the following when no editor is specified:

- Use `xdg-open` on Linux.
- Use `open` on macOS. 
- Use `start` on Windows.

All of the above are the standard ways of opening files using the associated program.

### PR DESCRIPTION
**Pull request by n0-fun**
This is tested working on Linux, Windows, and macOS. Closes #6

### CODE REVIEW

**Comment by n0-fun** [comment]
This is confirmed working on macOS. Closes #6 .

File: RegexFileSearcher/RegexFileSearcher/FileHandler.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Text;
```
Comment (n0-fun) [inline RegexFileSearcher/RegexFileSearcher/FileHandler.cs:5]:
```suggestion
using System.Diagnostics;
using System.Runtime.InteropServices;
```
Unnecessary `using` directives.

File: RegexFileSearcher/RegexFileSearcher/FileHandler.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+namespace RegexFileSearcher
+{
+    class FileHandler
```
Comment (n0-fun) [inline RegexFileSearcher/RegexFileSearcher/FileHandler.cs:9]:
```suggestion
    internal class FileHandler
```
Explicitly adding the `internal` accessibility modifier here makes it consistent with the other classes and also satisfies the  analyzers.

File: RegexFileSearcher/RegexFileSearcher/FileHandlerException.cs [hunk 3]
```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
```
Comment (n0-fun) [inline RegexFileSearcher/RegexFileSearcher/FileHandlerException.cs:3]:
```suggestion
﻿using System;
```
Unnecessary `using` directives.

File: RegexFileSearcher/RegexFileSearcher/FileHandlerException.cs [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+{
+    [Serializable]
+    public class FileHandlerException : Exception
+    {
+        public FileHandlerException() { }
+        public FileHandlerException(string message) : base(message) { }
+        public FileHandlerException(string message, Exception inner) : base(message, inner) { }
+        protected FileHandlerException(
+          System.Runtime.Serialization.SerializationInfo info,
+          System.Runtime.Serialization.StreamingContext context) : base(info, context) { }
```
Comment (n0-fun) [inline RegexFileSearcher/RegexFileSearcher/FileHandlerException.cs:15]: Any particular reason for not `using` the `System.Runtime.Serialization` namespace?

**Comment by n0-fun** [comment]
@CommonLoon102 The code is still working as before. I've added some cleanup code you might want to check out.
By the way, wouldn't a static class + static constructor make more sense for `FileHandler`?

File: RegexFileSearcher/RegexFileSearcher/FileHandlerException.cs (hunk 4 above)
Comment (CommonLoon102) [reply RegexFileSearcher/RegexFileSearcher/FileHandlerException.cs:15]: No, I was just writing `ex` `tab` `tab` `FileHandler` `enter` and that's what it generated.

**Pull request merged**